### PR TITLE
1042 - update ids-list-view-item (added ids-list-view-item child component)

### PR DESCRIPTION
**Pull request by clayinfor**
**Explain the details for making this change. What existing problem does the pull request solve?**

This PR adds support for `ids-list-view-item` child component to IdsListView.

```html
<ids-list-view selectable="multiple" sortable>
  <ids-list-view-item>
    <ids-text font-size="16" type="h2">First Column</ids-text>
    <ids-text font-size="12" type="span">ID: 1</ids-text>
    <ids-text font-size="12" type="span">Comments: One</ids-text>
  </ids-list-view-item>
  <ids-list-view-item active>
    <ids-text font-size="16" type="h2">Second Column</ids-text>
    <ids-text font-size="12" type="span">ID: 2</ids-text>
    <ids-text font-size="12" type="span">Comments: Two</ids-text>
  </ids-list-view-item>
  <ids-list-view-item disabled>
    <ids-text font-size="16" type="h2">Third Column</ids-text>
    <ids-text font-size="12" type="span">ID: 3</ids-text>
    <ids-text font-size="12" type="span">Comments: Three</ids-text>
  </ids-list-view-item>
</ids-list-view>
```

**Related github/jira issue (required)**:

Closes #1042 

**Steps necessary to review your pull request (required)**:

1. Check out branch and run: `nvm use && npm run start`
2. Check http://localhost:4300/ids-list-view/list-view-items.html
4. Check http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html
5. Check http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html
6. Check http://localhost:4300/ids-list-view/list-view-items-search.html
7. Check http://localhost:4300/ids-list-view/list-view-items-sortable.html
8. Run tests to ensure they all pass: `npx jest -- list-view`

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 - `[Tooltip]` Changed the tooltip heights to match. ([#7509](https://github.com/infor-design/enterprise/issues/7509))
 - `[Themes]` Added theme switcher to side-by-side examples and ability to switch 4.x themes in the `ids-theme-switcher `component. ([#939](https://github.com/infor-design/enterprise-wc/issues/939))
 - `[DataGrid]` Fixed a bug in the filter header where text selection in the inputs would cause accidental dragging. ([#1321](https://github.com/infor-design/enterprise-wc/issues/1321))
+- `[ListView]` Added support for `ids-list-view-item` child component. ([#1042](https://github.com/infor-design/enterprise-wc/issues/1042))
```
Comment (tmcconechy) [inline doc/CHANGELOG.md:23]: Alpha. Order please

File: src/components/ids-list-view/ids-list-view-item.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import '../ids-swappable/ids-swappable';
+import '../ids-swappable/ids-swappable-item';
+import styles from './ids-list-view-item.scss';
+
+const Base = IdsEventsMixin(
+  IdsElement
+);
+
+/**
+ * IDS List View Component
```
Comment (tmcconechy) [inline src/components/ids-list-view/ids-list-view-item.ts:16]:
```suggestion
 * IDS List View Item Component
```

File: src/components/ids-list-view/ids-list-view-item.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  set active(value: boolean) {
+    if (stringToBool(value)) {
+      this.setAttribute(attributes.ACTIVE, 'true');
+    } else {
+      this.removeAttribute(attributes.ACTIVE);
+    }
+  }
+
+  set disabled(value: boolean) {
```
Comment (tmcconechy) [inline src/components/ids-list-view/ids-list-view-item.ts:79]: Can we add JS Doc for all these setters?

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Everything looks good but a few trivial doc/ suggestions inline.

Also on some of the examples im wondering about a few things

1) on http://localhost:4300/ids-list-view/list-view-items.html i see two selected but i cant seem to change selection. Maybe just remove selection from the example page since its on another page?
2) similarly on http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html i cant seem to "activate" the rows (change by clicking)
3) http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html also a problem changing selection
4) on http://localhost:4300/ids-list-view/list-view-items-search.html a) the search doesnt seem to work and b) it would look better if it was 100% width

Otherwise really nice work here.

File: doc/CHANGELOG.md [hunk 4]
```diff
@@ -17,9 +17,10 @@
 - `[DataGrid]` Add ability to multi select with shift key. ([#1330](https://github.com/infor-design/enterprise-wc/issues/1330))
 - `[DataGrid]` Fix tree not selecting or collapsing all children in AngularJS. ([#1284](https://github.com/infor-design/enterprise-wc/issues/1284))
 - `[DataGrid]` Fix tree collapse/expand state while sorting. ([#1284](https://github.com/infor-design/enterprise-wc/issues/1284))
-- `[Tooltip]` Changed the tooltip heights to match. ([#7509](https://github.com/infor-design/enterprise/issues/7509))
-- `[Themes]` Added theme switcher to side-by-side examples and ability to switch 4.x themes in the `ids-theme-switcher `component. ([#939](https://github.com/infor-design/enterprise-wc/issues/939))
 - `[DataGrid]` Fixed a bug in the filter header where text selection in the inputs would cause accidental dragging. ([#1321](https://github.com/infor-design/enterprise-wc/issues/1321))
+- `[ListView]` Added support for `ids-list-view-item` child component. ([#1042](https://github.com/infor-design/enterprise-wc/issues/1042))
```
Comment (tmcconechy) [inline doc/CHANGELOG.md:21]: Can you add this in new section `### 1.0.0-beta.14 Fixes` now (depending if that section is there yet if you merge main)

**Comment by EdwardCoyle** [comment]
Overall everything looks great with this.  However, I'm seeing something when the main example page loads -- seems like one in every five-or-so refreshes causes the List View to populate with no actual contents, and more rows.  Most of the time, the correct behavior occurs.  I managed to catch it a couple times in the video below:

[Untitled_ Jul 17 2023 9_31 AM.webm](https://github.com/infor-design/enterprise-wc/assets/3249601/4700f8e7-046a-4a9f-b4ad-8c5fa40f0ed8)

**Comment by tmcconechy** [comment]
@clayinfor put WIP on as i keep checking this but appears incomplete from some comments. Just remove the tag when your done 👍🏻

**Comment by clayinfor** [comment]
Hey @tmcconechy , I removed the WIP label from this as it's good to go... I was actually working on an almost total refactor but I'm running out of time in the sprint... so I will have to circle back to it later.  

The issue @EdwardCoyle pointed out seems to happen for me in incognito mode and I think it has something to do with the JSON file not being asynchronously loaded sometimes. Will have to look into that later as well.

**Comment by clayinfor** [comment]
hey @tmcconechy , I'm thinking we merge this one, and then I'll circle back to the other PR to get that merged later.

**Comment by tmcconechy** [comment]
@clayinfor i still see all of these issues:

1. on http://localhost:4300/ids-list-view/list-view-items.html i see two selected but i cant seem to change selection. Maybe just remove selection from the example page since its on another page?
2. similarly on http://localhost:4300/ids-list-view/list-view-items-selectable-mixed.html i cant seem to "activate" the rows (change by clicking)
3. http://localhost:4300/ids-list-view/list-view-items-selectable-multiple.html also a problem changing selection
4. on http://localhost:4300/ids-list-view/list-view-items-search.html a) the search doesnt seem to work and b) it would look better if it was 100% width

So basically search and selection dont work anymore.
